File: solutions/230.KthSmallestElementinaBST/dj.cpp
#include <vector>
using namespace std;

  struct TreeNode {
      int val;
      TreeNode *left;
      TreeNode *right;
      TreeNode() : val(0), left(nullptr), right(nullptr) {}
      TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
      TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
  };

class Solution {
public:
    int kthSmallest(TreeNode* root, int k) {
        vector<int> increasing;
        inorder(root, increasing);
        return increasing[k-1];
    }
    void inorder(TreeNode* root, vector<int>& traversal) {
        if (root) {
            inorder(root->left, traversal);
            traversal.push_back(root->val);
            inorder(root->right, traversal);
        }
    }
};

File: solutions/129.SumRoottoLeafNumbers/dj.py
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        ans = 0

        def addSum(node, cur):
            nonlocal ans
            cur = cur * 10 + node.val
            if not node.left and not node.right:
                ans += cur
                return
            if node.left:
                addSum(node.left, cur)
            if node.right:
                addSum(node.right, cur)
        
        addSum(root, 0)
        return ans